calculator_submit.rb

```diff
@@ -0,0 +1,58 @@
+# this version is an experiment with while loops and the .include method
+puts "Hi! Let's do math!"
+
+puts "Please tell me which kind of math you would like to do"
+math_type = gets.chomp.downcase
+
+#this array includes all valid math types
+#the while loop loops through the array with the .include method with the
+#argument of math_type as the paramenter
+#as long as the array.include?(math_type) is false the invalid input message
+#will print to the screen
+valid_math_types = ['+', 'add', 'addition', '-', 'sub', 'subtraction', '*',
+  'mult', 'multiply', 'multiplication', '/', 'div', 'divide', 'division', '^',
+  'exponent', 'exponential', '%', 'modulo', 'mod']
+  while valid_math_types.include?(math_type) == false
+    puts "Please insert a valid math operation"
+    math_type = gets.chomp.downcase
+  end
+
+puts "Great! Please enter the first number for #{math_type}."
+first_number = gets.chomp.to_i
+#Consider checking for invalid string input THEN changing to an integer
```

Comment:
This is a great question! and it's definitely a hard thing to do in a really readable way, IMO.

My leaning is to go the way of using a regular expression here (even though my philosophy is to use them sparingly, since they aren't typically readable, it's definitely a last resort).  This whole stackoverflow post (http://stackoverflow.com/questions/5661466/test-if-string-is-a-number-in-ruby-on-rails) has a whole bunch of ways to catch these things based on what you want... if you just want positive integers to be allowed, it would be `my_string.should =~ /^[0-9]+$/`. (that's not very pretty, don't you wish ruby had a function that did this kind of check, `is_numeric?` or something?)

Another clever thing, which isn't very readable, either, in my opinion, is to note that `to_i` always returns 0 if the string in question is not a numeric value.  If you used that knowledge, however, you'd have to figure out a way to catch the edge cases where the user actually enters `0` (or `0000`, or `0.0` or `0.1`.... etc...).  In that case, you might end up using a regular expression after all for that edge case.